Packages/io.chainsafe.web3-unity/Runtime/Scripts/WalletConnect/WCUnityLogger.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using UnityEngine;
+using ILogger = WalletConnectSharp.Common.Logging.ILogger;
+
+namespace Web3Unity.Scripts.WalletConnect
+{
+    public class WCUnityLogger : ILogger
+    {
+        public void Log(string message)
+        {
+            MainThreadDispatcher.Enqueue(() => Debug.Log(message));
```

Comment:
I'll take a look, I'm not sure it'll fit but if it does I'll refactor it in

Reply:
Yeah, give it a try. I think it's good to keep all logs coming through one funnel.

Reply:
this is the only way I could find, it's a very long process if there's another way let me know. We could have `WCLogger` implement `ILogWritter` but that's sort of irrelevant since `WalletConnectSharp.Common.Logging.ILogger` already has `ILogWritter`'s methods

Reply:
I would pass ILogWriter reference to WCUnityLogger and delegate it's calls to ILogWriter like this:
`public void LogError(string message) => logWriter.LogError(message);`

Reply:
done